test/test_e2e.rb

```diff
@@ -129,4 +129,10 @@ def test_suite_with_failures_and_fail_fast
 
     assert_includes [2, 3], queue.processed_jobs.length
   end
+
+  def test_suite_with_rspec_arguments
+    queue = exec_build("tagged_suite", "-- --tag foo")
+
+    assert_equal 1, queue.example_count
+  end
```

Comment:
To be on the safe side, I think we should test for more things here. A few thoughts:

1. we could have the "bar" example fail, and assert that the build is still successful in the end (i.e. the `bar` example was filtered out).
2. we could assert the actual contents of `queue.processed_jobs`
3. we could cover more surface area by using a more complex combination of flags, for example: `--tag foo --format progress --format documentation --output foo.txt --no-color --backtrace --profile 2 spec/tagged_spec.rb`